cinema/serializers.py

```diff
@@ -68,13 +79,66 @@
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available"
         )
 
 
+class TicketSeatRowSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("row", "seat",)
+
+
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = TicketSeatRowSerializer(
+        many=True,
+        read_only=True,
+        source="tickets",
+    )
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = (
+            "id",
+            "show_time",
+            "movie",
+            "cinema_hall",
+            "taken_places",
+        )
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    movie_session = MovieSessionListSerializer(many=False, read_only=True)
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+
+    def validate(self, data):
+        data = super(TicketSerializer, self).validate(data)
+
+        ticket = Ticket(
+            movie_session=data.get("movie_session"),
+            row=data.get("row"),
+            seat=data.get("seat"),
+        )
+        ticket.full_clean()
```

Comment:
The `full_clean()` method in `TicketSerializer`'s `validate` method will raise a `ValidationError` if the ticket is invalid. Ensure that this is the intended behavior, as it will prevent saving invalid tickets. Consider handling the exception to provide more user-friendly error messages if needed.